File: IAIso-v5.0/core/iaiso-ruby/lib/iaiso/core/lifecycle.rb
# frozen_string_literal: true

module IAIso
  module Core
    # Lifecycle states for a PressureEngine. The string value is the wire form.
    module Lifecycle
      INIT      = "init"
      RUNNING   = "running"
      ESCALATED = "escalated"
      RELEASED  = "released"
      LOCKED    = "locked"

      ALL = [INIT, RUNNING, ESCALATED, RELEASED, LOCKED].freeze
    end
  end
end

File: IAIso-v5.0/core/iaiso-php/src/Core/Lifecycle.php
<?php

declare(strict_types=1);

namespace IAIso\Core;

/** Lifecycle states for a {@see PressureEngine}. Wire form is lowercase. */
enum Lifecycle: string
{
    case Init       = 'init';
    case Running    = 'running';
    case Escalated  = 'escalated';
    case Released   = 'released';
    case Locked     = 'locked';
}
